viewing_party/party.py

```diff
@@ -1,23 +1,231 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+
+    #return None of title, genre, and rating are empty
+    if not (title and genre and rating):
+        return None
+    
+    movie = {
+                "title": title,
+                "genre": genre,
+                "rating": rating
+            
+            }
+    return movie
+
+
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+    
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+    
+    return user_data
+
+
+def watch_movie(user_data, title): 
+
+    #loop through each movie in the watchlist
+    for movie in user_data["watchlist"]:
+
+        #if it matches the title passed as the arg add it to watch
+        if title == movie["title"]:
+
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+        
+    return user_data
+            
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):

```

Comment:
We want to group related code together in a function physically and visually. I would strongly suggest removing most new lines as part of your clean up step, leaving lines between your functions and a few inside your functions if they break up logical sections of the code.